src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionTests_WithElement_Extra.cs

```diff
@@ -2427,40 +3279,37 @@ class MyBuilder
                 c = [with(out r), 3];
                 c.Report();
                 """;
-        var comp = CreateCompilation(
+        // Added execution output verification.
+        var verifier = CompileAndVerify(
             [sourceA, sourceB1, s_collectionExtensions],
-            targetFramework: TargetFramework.Net80);
-        comp.VerifyEmitDiagnostics(
-            // (5,6): error CS9502: Collection arguments are not supported for type 'MyCollection<int>'.
-            // c = [with(out x)];
-            Diagnostic(ErrorCode.ERR_CollectionArgumentsNotSupportedForType, "with").WithArguments("MyCollection<int>").WithLocation(5, 6),
-            // (8,6): error CS9502: Collection arguments are not supported for type 'MyCollection<int>'.
-            // c = [with(out r), 3];
-            Diagnostic(ErrorCode.ERR_CollectionArgumentsNotSupportedForType, "with").WithArguments("MyCollection<int>").WithLocation(8, 6));
+            targetFramework: TargetFramework.Net80,
+            expectedOutput: IncludeExpectedOutput("[0], [3, 0], "),
+            verify: Verification.FailsPEVerify);
+        verifier.VerifyDiagnostics();
 
         string sourceB2 = """
-#pragma warning disable 219 // variable assigned but never used
+                #pragma warning disable 219 // variable assigned but never used
                 MyCollection<int> c;
                 int x = 1;
                 c = [with(1)];
                 c = [with(x)];
                 c = [with(ref x)];
                 c = [with(in x)];
                 """;
-        comp = CreateCompilation([sourceA, sourceB2], targetFramework: TargetFramework.Net80);
+        var comp = CreateCompilation([sourceA, sourceB2], targetFramework: TargetFramework.Net80);
         comp.VerifyEmitDiagnostics(
-            // (4,6): error CS9502: Collection arguments are not supported for type 'MyCollection<int>'.
+            // (4,11): error CS1620: Argument 1 must be passed with the 'out' keyword
             // c = [with(1)];
-            Diagnostic(ErrorCode.ERR_CollectionArgumentsNotSupportedForType, "with").WithArguments("MyCollection<int>").WithLocation(4, 6),
-            // (5,6): error CS9502: Collection arguments are not supported for type 'MyCollection<int>'.
+            Diagnostic(ErrorCode.ERR_BadArgRef, "1").WithArguments("1", "out").WithLocation(4, 11),
```

Comment:
Just some trivia, consider testing, it should be possible to invoke a `Create` method with `ref` parameters without using `ref` arguments, if its containing type is a ComImport type. See `AllowRefOmittedArguments`